Index: Mesa/src/mesa/drivers/x11/xm_image.c
===================================================================
--- Mesa.orig/src/mesa/drivers/x11/xm_image.c
+++ Mesa/src/mesa/drivers/x11/xm_image.c
@@ -49,7 +49,7 @@ XMesaImage *XMesaCreateImage(int bitsPer
 {
     XMesaImage *image;
 
-    image = (XMesaImage *)xalloc(sizeof(XMesaImage));
+    image = (XMesaImage *)malloc(sizeof(XMesaImage));
 
     if (image) {
 	image->width = width;
@@ -67,7 +67,7 @@ void XMesaDestroyImage(XMesaImage *image
 {
     if (image->data)
 	free(image->data);
-    xfree(image);
+    free(image);
 }
 
 unsigned long XMesaGetPixel(XMesaImage *image, int x, int y)
